hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/LazyDefiningSpecificMemberPms.java

```diff
@@ -0,0 +1,154 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl.connector.map;
+
+import com.hazelcast.cluster.Address;
+import com.hazelcast.function.SupplierEx;
+import com.hazelcast.jet.JetException;
+import com.hazelcast.jet.core.ProcessorMetaSupplier;
+import com.hazelcast.jet.core.ProcessorSupplier;
+import com.hazelcast.jet.core.test.TestProcessorMetaSupplierContext;
+import com.hazelcast.jet.impl.execution.init.Contexts;
+import com.hazelcast.jet.sql.impl.JetSqlSerializerHook;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.security.PermissionsUtil;
+import com.hazelcast.sql.impl.expression.Expression;
+import com.hazelcast.sql.impl.expression.ExpressionEvalContext;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.function.Function;
+
+public class LazyDefiningSpecificMemberPms extends ProcessorMetaSupplier.SpecificMemberPms {
+
+    Map<Address, int[]> partitionAssignment;
+    int partitionId;
+
+    // TODO: transient?
+    private Integer partitionArgIndex;
+    private SupplierEx<?> partitionKeyExprSupplier;
+
+    public LazyDefiningSpecificMemberPms() {
+        super();
+    }
+
+    private LazyDefiningSpecificMemberPms(ProcessorSupplier supplier, int partitionArgumentIndex) {
+        this.supplier = supplier;
+        this.partitionArgIndex = partitionArgumentIndex;
+        this.memberAddress = null;
+    }
+
+    private LazyDefiningSpecificMemberPms(ProcessorSupplier supplier, SupplierEx<?> partitionKeyExprSupplier) {
+        this.supplier = supplier;
+        this.partitionKeyExprSupplier = partitionKeyExprSupplier;
+        this.memberAddress = null;
+    }
+
+    @Override
+    public void init(@Nonnull Context context) throws Exception {
+        PermissionsUtil.checkPermission(supplier, context);
+        if (context.localParallelism() != 1) {
+            throw new IllegalArgumentException(
+                    "Local parallelism of " + context.localParallelism() + " was requested for a vertex that "
+                            + "supports only total parallelism of 1. Local parallelism must be 1.");
+        }
+
+        ExpressionEvalContext eec = ExpressionEvalContext.from(context);
+        Expression<?> partitionKeyExpr = null;
+        if (partitionKeyExprSupplier != null) {
+            Object obj = partitionKeyExprSupplier.get();
+            if (!(obj instanceof Expression)) {
+                throw new RuntimeException("");
+            }
+            partitionKeyExpr = (Expression<?>) obj;
+        }
+
+        // test path
+        if (context instanceof TestProcessorMetaSupplierContext) {
```

Comment:
`Accessors` can obtain nodeEngine from HazelcastInstance, special test path should not be needed

Reply:
#24908